### DISCUSSION TITLE
Git Action Workflows - Ability to invoke another step in the middle of a parent step

### DISCUSSION THREAD

**Discussion by ppmgitrepo** [PrestaShop/PrestaShop#35808] [Ideas]
### The problem

In a Git Actions Workflow 
I have 2 steps 

Step1 
Step2 

Step1 processes many rows in an iterative While .. Loop
from within the loop, I want to have the ability to invoke Step2 , execute it and return back to Step1 to the line where it was invoked, and continue processing the rest of the lines in Step1 within the loop.
(because the code in Step2 just does some iterative task) 

It is like a Function() invocation call within the loop. 

### The solution

It will make coding much easier when you have to do bulk processing. 
You can split the processing into multiple steps that does smaller repetitive tasks and just invoke them in the main Step. 

### Alternatives

no direct alternatives. 

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
Hello @ppmgitrepo are you sure you created that item on the right repository? This is the PrestaShop project and we do not control the GitHub Actions workflow

**Reply by matks** [reply]
I think you should rather write your discussion here https://github.com/orgs/community/discussions

**Comment by ppmgitrepo** [comment]
Sorry my bad - closing this.
